### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by perebalazs** [spin83/multi-monitors-add-on#72]
When this extension is turned on, the memory usage of gnome-shell increases. The highest level which I experienced was ~2GB. The increase is happening in steps, eg. when I launch a program, the memory usage of gnome-shell jumps from ~170MB to ~250MB and so on. If this extension is turned off, there is no such high jumps in memory usage.

Ubuntu: 18.04
gnome-shell: 3.28.3-0ubuntu0.18.04.2
gjs version: 1.53.3-1

**Comment by shadyyx** [comment]
I can second to this. Did not inspect the exact behavior, but can confirm that my Ubuntu has frozen already one time and had to be hard restarted. I did a research and read about gnome memory leak bug which is still present even in gnome shell 3.28 and that ALT+F2 followed by R and ENTER fixes this temporarily.

During the trial I have found out that the memory didn't drop as was expected and that this extension was just turned off with an unspecified error icon and report both in Gnome Tweaks and at extensions.gnome.org. After uninstalling the extension the memory dropped drastically about 2~3 GB! After installing the extension again, it seems to be turned on and works, despite the same error icon is still present.

Ubuntu 18.04
gnome shell 3.28.3
gjs 1.52.1
